src/uart/registers.rs

```diff
@@ -177,4 +177,127 @@ impl<const PORT_PTR: usize> ControlRegister<PORT_PTR> {
     /// # Activate Receive FIFO Flush
     /// Write a 1 to flush the receive FIFO
     activate_receive_fifo_flush}
+
+    bit_impl! {8, RESET, // FIXME This need to be renamed / changed
+    /// # Activate Transmit FIFO Flush
+    /// Write a 1 to flush the transmit FIFO
+    activiate_transmit_fifo_flush}
+
+    bit_impl! {7, RW,
+    /// # Set `CTS` Sampling Disable
+    /// Choose to enable or disable `CTS` (Clear To Send)
+    /// - 0: Enabled
+    /// - 1: Disabled
+    set_cts_sampling_disable,
+    /// # Check `CTS` Sampling Disable
+    /// Check the current state of CT disable
+    /// - 0: Enabled
+    /// - 1: Disabled
+    check_cts_sampling_disable}
+
+    bit_impl! {6, RW,
+    /// # Set Parity Value
+    set_parity_value,
+    /// # Check Parity Value
+    check_parity_value}
+
+    bit_impl! {5, RW,
+    /// # Parity Odd Even Select
+    set_parity_odd_even,
+    /// # Check Parity Odd Even Select
+    check_parity_odd_even}
+
+    bit_impl! {4, RW,
+    /// # Set Transmit Parity Generation Enable
+    set_transmit_parity_generation_enable,
+    /// # Check Transmit Parity Generation Enable
+    check_transmit_parity_genration_enable}
+
+    bit_impl! {0..=3, RW u8,
+    /// # Set Receive FIFO Threshold
+    set_recieve_fifo_threshold,
+    /// # Check Receive FIFO Threshold
+    check_recieve_fifo_threshold}
+}
+
+/// # UART Status Register
+/// The UART Status Register. See page 182, table 12-9
+pub struct StatusRegister<const PORT_PTR: usize> {}
+reg_impl!(RW, StatusRegister, uro::UART_CTRL);
```

Comment:
This can be made `RO` since all of its fields only use `RO`. 
```diff
- reg_impl!(RW, StatusRegister, uro::UART_CTRL);
+ reg_impl!(RO, StatusRegister, uro::UART_CTRL);
```